### ISSUE TITLE
b-overlay does not obscure b-modal

### ISSUE THREAD

**Issue by ahurlburt** [bootstrap-vue/bootstrap-vue#6774]
When adding a b-overlay outside a b-modal (and setting a high z-index) the modal content is not obscured.

The spinner is present but the height of the overlay backdrop is 0. This is not desired.

It seems like a common use case to be able to obscure a modal if it calls an API or something yet I see no examples of this.

This code sample causes the issue

```
<b-overlay :show="requestInProgress" z-index="2000">
<b-modal v-model="showModal" size="lg" title="Modal Title" hide-footer>

.. content in modal ..

</b-modal>
</b-overlay>

```

if I set `no-wrap` and `fixed` on the overlay the backdrop will obscure everything however for some reason it blocks all input on the page even when `show` is false.

Is this a bug or am I doing something incorrectly?

Thank you,

You can see the spinner being displayed over the modal content here but you can see the modal content clearly, nothing is obscured by backdrop which is not desirable.

![image](https://user-images.githubusercontent.com/16481642/132921116-27952899-b597-4c93-b55e-0b985f689c64.png)

**Comment by Hiws** [comment]
The modal is appended to `<body>` by default, which means it wont be rendered inside the overlay.

Have you thought about placing the overlay *inside* the modal instead and using the `no-wrap` prop?
```html
<b-modal size="lg" title="Modal Title" hide-footer>
  <b-overlay no-wrap></b-overlay>
  
  Modal content
</b-modal>
```
![XGT9tfk 1](https://user-images.githubusercontent.com/7057431/132947863-d2a480e3-7623-470a-b19b-fdded44b08f3.png)

**Comment by ahurlburt** [comment]
Thank you @Hiws 

I did try placing inside the modal using the `no-wrap` but it wasn't obscuring the full modal (title was still visible).

One thing I didn't mention but was maybe relevant was I had an overlay with custom content. 

I got it to work eventually by placing the `<b-overlay> <template #overlay> ... </template></b-overlay>` at the end of the modal definition right before the closing `<b-modal>` tag and setting the `no-wrap` and `fixed` properties. I removed the `z-index`.

I guess i can close this though I'm still not sure why I was seeing issues with original code..
